core/domain/user_services.py

```diff
@@ -656,3 +668,40 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+
+def get_weekly_dashboard_stats(user_id):
+    """Returns a list which contains the dashboard stats of a user,
```

Comment:
I think the docstring here could be clearer. Specify the format of the return value explicitly, and what the values of the keys mean; specify what happens in the error cases (i.e., user doesn't exist).